### ISSUE TITLE
Support for "dumb" remote access

### ISSUE THREAD

**Issue by jelmer** [jelmer/dulwich#1097]
This is probably a matter of implementing a DumbRemoteRepo that implements enough of the API to support fetch_pack.

Needs a custom object store implementation that just reads a generated pack info file.

**Comment by jelmer** [comment]
I've got an initial implementation of this.

**Comment by nanonyme** [comment]
@jelmer we are hitting this quite a lot. Did that initial implementation ever turn into something more mergeable?

**Comment by jelmer** [comment]
Not much progress unfortunately. Where are you hitting this? Dumb repos are fairly rare nowadays.

**Comment by nanonyme** [comment]
There's plenty of self-hosting projects using dumb repos including gpgme,  libnftl and various others.

**Comment by nanonyme** [comment]
Sure, it's by far minority of total projects we package but still significant enough for this feature to be relevant.

**Comment by SanitizedMemory** [comment]
+1 I am hitting the initial issue as well. Trying to run `poetry install`, where one of the package dependencies is a local (private) github repo. Works fine in my local env, but fails inside the Dockerfile.

**Comment by nanonyme** [comment]
> +1 I am hitting the initial issue as well. Trying to run `poetry install`, where one of the package dependencies is a private github repo. Works fine in my local env, but fails inside the Dockerfile. I have verified that my ssh keys are being passed correctly.

That sounds odd. Using GitHub should not result in dumb repo whether you are using private repos or not.

**Comment by SanitizedMemory** [comment]
> > +1 I am hitting the initial issue as well. Trying to run `poetry install`, where one of the package dependencies is a private github repo. Works fine in my local env, but fails inside the Dockerfile. I have verified that my ssh keys are being passed correctly.
> 
> That sounds odd. Using GitHub should not result in dumb repo whether you are using private repos or not.

You're right -- I fell down the wrong rabbit hole. Fixed my issue and it was unrelated to this. Thanks!
